src/filters/simple-filter-directive.js

```diff
@@ -183,13 +183,23 @@ angular.module('patternfly.filters').directive('pfSimpleFilter', function () {
         return foundFilter !== undefined;
       };
 
+      $scope.enforceSingleSelect = function (filter) {
+        if (filter.type === 'select') {
+          _.remove($scope.config.appliedFilters, {title: filter.title, type: filter.type});
+        }
+      };
+
       $scope.addFilter = function (field, value) {
         var newFilter = {
           id: field.id,
           title: field.title,
+          type: field.filterType,
           value: value
         };
         if (!$scope.filterExists(newFilter)) {
+
+          $scope.enforceSingleSelect(newFilter);
+
           $scope.config.appliedFilters.push(newFilter);
 
```

Comment:
I think Dave's idea of doing a remove and add rather than a replace makes sense. The "jolt" in the filter list should make it more obvious that something has changed and that the old filter is no longer active.